### PR TITLE
SB-805: Migrate from EHCache to Hazelcast [obsolete]

### PR DESCRIPTION
**Pull request by carlspring**

### CODE REVIEW

**Comment by carlspring** [comment]
Thanks for your help @sbespalov!

**Comment by carlspring** [comment]
Feedback from @sbespalov :
> Sergey, 5:56 PM
> cahce should be synchronized with transactions
> 
> there was `EhCacheCacheManager.setTransactionAware(true)` for ehcahce but seems that it's not so simple for Hazelcast
> 
> they have separate `HazelcastTransactionManager` which need to be manually synchronized with `JpaTransactionManager`
> 
> probably it can be done with XA transactions
> 
> anyway some aditional configuration will be required for this

**Comment by carlspring** [comment]
A potentially useful pull request: https://github.com/hazelcast/hazelcast/pull/7469.

File: strongbox-security/strongbox-user-management/src/main/java/org/carlspring/strongbox/users/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -30,6 +33,7 @@
  *
  * @author Alex Oreshkevich
  */
+@CacheConfig(cacheNames = "users")
```
[the remark this answers is not in the document]
Comment (carlspring) [reply strongbox-security/strongbox-user-management/src/main/java/org/carlspring/strongbox/users/service/impl/UserServiceImpl.java]: Noted!

File: strongbox-data-service/src/main/java/org/carlspring/strongbox/config/DataServiceConfig.java [hunk 2]
```diff
@@ -128,30 +109,44 @@ public OEntityManager oEntityManager()
...
+    public CacheManager cacheManager()
+    {
+        return new HazelcastCacheManager(hazelcastInstance());
+    }
+
+    @Bean
+    public HazelcastTransactionManager transactionManager()
+    {
+        HazelcastTransactionManager hazelcastTransactionManager = new HazelcastTransactionManager(hazelcastInstance());
+        hazelcastTransactionManager.setTransactionSynchronization(SYNCHRONIZATION_ON_ACTUAL_TRANSACTION);
```
Comment (carlspring) [inline strongbox-data-service/src/main/java/org/carlspring/strongbox/config/DataServiceConfig.java]: @sbespalov , @fuss86 ,

Would you mind having a look at this? There are three possible settings for the transaction synchronization. Based on @sbespalov 's comments, I assumed this is the correct one to use. However, this is now causing:

```
03:05:07.099 20-04-2018 | DEBUG | main                 | o.c.s.security.vote.MethodAccessDecisionManager    | Voter: org.carlspring.strongbox.security.vote.ExtendedAuthoritiesVoter@696b4a95, returned: 1
03:05:07.099 20-04-2018 | DEBUG | main                 | c.h.spring.transaction.HazelcastTransactionManager | Found thread-bound TransactionContext [com.hazelcast.transaction.impl.TransactionContextImpl@74723d09]
03:05:07.099 20-04-2018 | DEBUG | main                 | c.h.spring.transaction.HazelcastTransactionManager | Participating in existing transaction
03:05:07.099 20-04-2018 | DEBUG | main                 | c.h.spring.transaction.HazelcastTransactionManager | Found thread-bound TransactionContext [com.hazelcast.transaction.impl.TransactionContextImpl@74723d09]
03:05:07.099 20-04-2018 | DEBUG | main                 | c.h.spring.transaction.HazelcastTransactionManager | Participating in existing transaction
03:05:07.211 20-04-2018 | DEBUG | main                 | c.h.spring.transaction.HazelcastTransactionManager | Found thread-bound TransactionContext [com.hazelcast.transaction.impl.TransactionContextImpl@74723d09]
03:05:07.212 20-04-2018 | DEBUG | main                 | c.h.spring.transaction.HazelcastTransactionManager | Participating in existing transaction
03:05:07.215 20-04-2018 | DEBUG | main                 | o.s.w.s.m.m.annotation.HttpEntityMethodProcessor   | Written [org.carlspring.strongbox.controllers.support.ResponseEntityBody@4fed6dd1] as "application/json" using [org.springframework.http.converter.json.MappingJackson2HttpMessageConverter@c1b96e6]
200
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Pragma: no-cache
Expires: 0
X-Frame-Options: DENY
Content-Type: application/json;charset=UTF-8

{
  "message" : "The user was updated successfully."
}
03:05:07.217 20-04-2018 | DEBUG | main                 | c.h.spring.transaction.HazelcastTransactionManager | Found thread-bound TransactionContext [com.hazelcast.transaction.impl.TransactionContextImpl@74723d09]
03:05:07.217 20-04-2018 | DEBUG | main                 | c.h.spring.transaction.HazelcastTransactionManager | Participating in existing transaction
03:05:07.342 20-04-2018 | DEBUG | main                 | c.h.spring.transaction.HazelcastTransactionManager | Found thread-bound TransactionContext [com.hazelcast.transaction.impl.TransactionContextImpl@74723d09]
03:05:07.342 20-04-2018 | DEBUG | main                 | c.h.spring.transaction.HazelcastTransactionManager | Suspending current transaction, creating new transaction with name [null]
03:05:07.343 20-04-2018 | DEBUG | main                 | c.h.spring.transaction.HazelcastTransactionManager | Initiating transaction rollback
03:05:07.343 20-04-2018 | DEBUG | main                 | c.h.spring.transaction.HazelcastTransactionManager | Rolling back Hazelcast transaction on TransactionContext [com.hazelcast.transaction.impl.TransactionContextImpl@74723d09]
[ERROR] Tests run: 37, Failures: 0, Errors: 35, Skipped: 0, Time elapsed: 3.682 s <<< FAILURE! - in org.carlspring.strongbox.controllers.users.UserControllerTestIT
[ERROR] userShouldNotBeAbleToChangeSomeoneElsePasswordToNullWithJsonAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.095 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] userWithoutViewUserRoleShouldNotBeAbleToViewOtherUserAccountData(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.038 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] testUpdateUserWithTextAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.212 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] greetTestWithJsonAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.01 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] userWithoutUpdateUserRoleShouldNotBeAbleToUpdateSomeoneElsePassword(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.013 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] userShouldBeAbleToChangeTheirOwnPasswordWithJsonAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.246 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] userShouldNotBeAbleToChangeTheirOwnPasswordToEmptyWithTextAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.183 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] shouldBeAbleToUpdateRolesWithJsonAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.245 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] testRetrieveAllUsers(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.047 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] userWithoutUsernameShouldNotBeAbleToUpdateWithTextAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.018 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] testCreateUserWithTextAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.026 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] shouldNotBeAbleToRetrieveUserThatNoExistsWithTextAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.013 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] userShouldNotBeAbleToChangeTheirOwnPasswordToNullWithJsonAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.016 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] userShouldNotBeAbleToChangeSomeoneElsePasswordToEmptyWithJsonAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.198 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] shouldNotBeAbleToCreateUserWithTheSameUsernameWithTextAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.034 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] testGetUser(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.02 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] userNotExistingShouldNotUpdateAccessModel(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.082 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] userShouldNotBeAbleToChangeSomeoneElsePasswordToNullWithTextAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.02 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] greetTestWithTextAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.024 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] testUpdateAccessModel(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.079 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] userShouldBeAbleToChangeTheirOwnPasswordWithTextAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.237 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] shouldBeAbleToUpdateRolesWithTextAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.236 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] testUpdateUserWithJsonAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.12 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] userShouldNotBeAbleToChangeTheirOwnPasswordToEmptyWithJsonAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.152 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] userShouldNotBeAbleToChangeTheirOwnPasswordToNullWithTextAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.019 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] notValidMapsShouldNotUpdateAccessModel(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.08 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] userShouldNotBeAbleToChangeSomeoneElsePasswordToEmptyWithTextAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.146 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] testDeleteUser(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.033 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] userWithoutUsernameShouldNotBeAbleToUpdateWithJsonAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.017 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] testCreateUserWithJsonAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.061 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] userWithoutViewUserRoleShouldBeAbleToViewHisAccountData(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.023 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] shouldNotBeAbleToRetrieveUserThatNoExistsWithJsonAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.016 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] userShouldNotBeAbleToDeleteHimself(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.015 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] shouldNotBeAbleToCreateUserWithTheSameUsernameWithJsonAcceptHeader(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.04 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

[ERROR] userWithoutUpdateUserRoleShouldBeAbleToUpdateHisPassword(org.carlspring.strongbox.controllers.users.UserControllerTestIT)  Time elapsed: 0.264 s  <<< ERROR!
org.springframework.transaction.TransactionSuspensionNotSupportedException: Transaction manager [com.hazelcast.spring.transaction.HazelcastTransactionManager] does not support transaction suspension
	at org.carlspring.strongbox.controllers.users.UserControllerTestIT.rollBackAdminUserPassword(UserControllerTestIT.java:71)

03:05:08.178 20-04-2018 | INFO  | Thread-5             | o.s.w.context.support.GenericWebApplicationContext | Closing org.springframework.web.context.support.GenericWebApplicationContext@4f2b503c: startup date [Fri Apr 20 03:04:21 BST 2018]; root of context hierarchy
03:05:08.181 20-04-2018 | INFO  | Thread-5             | org.quartz.core.QuartzScheduler                    | Scheduler schedulerFactoryBean_$_NON_CLUSTERED paused.
03:05:08.184 20-04-2018 | INFO  | Thread-5             | org.quartz.core.QuartzScheduler                    | Scheduler schedulerFactoryBean_$_NON_CLUSTERED shutting down.
03:05:08.185 20-04-2018 | INFO  | Thread-5             | org.quartz.core.QuartzScheduler                    | Scheduler schedulerFactoryBean_$_NON_CLUSTERED paused.
03:05:08.186 20-04-2018 | INFO  | Thread-5             | org.quartz.core.QuartzScheduler                    | Scheduler schedulerFactoryBean_$_NON_CLUSTERED shutdown complete.
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[ERROR]   UserControllerTestIT.rollBackAdminUserPassword:71 » TransactionSuspensionNotSupported
[INFO] 
[ERROR] Tests run: 102, Failures: 0, Errors: 35, Skipped: 0
```

Any ideas?

File: strongbox-data-service/src/main/java/org/carlspring/strongbox/config/DataServiceConfig.java [hunk 3]
```diff
@@ -97,15 +86,6 @@ public void init() throws ClassNotFoundException, LiquibaseException
             
             return null;
         });
-        
-
-    }
-
-    
-    @Bean
-    public PlatformTransactionManager transactionManager()
```
Comment (sbespalov) [inline strongbox-data-service/src/main/java/org/carlspring/strongbox/config/DataServiceConfig.java]: we shouldn't completely remove our DB transactions. I mean that we need to use `JtaTransactionManager` instead. *Hazelcast* supports JTA specification ([link](http://docs.hazelcast.org/docs/latest-development/manual/html/Transactions/Providing_XA_Transactions.html)). So probably we need somthing like implemented [here](https://github.com/hazelcast/hazelcast/issues/611#issuecomment-57178700).

File: strongbox-data-service/src/main/java/org/carlspring/strongbox/config/DataServiceConfig.java (hunk 2 above)
Comment (sbespalov) [reply strongbox-data-service/src/main/java/org/carlspring/strongbox/config/DataServiceConfig.java]: yeah, seems that Hazelcast don't support nesteed transactions. I am not sure would this affect on JTA (see my comments above), if so we will have to deal with this somehow, maybe use both JPA and JTA transactions (JPA as default and JTA where cache needed but with restriction on `REQUIRES_NEW` propagation).

**Comment by fuss86** [comment]
@sbespalov can't we re-implement `com.hazelcast.spring.cache.HazelcastCacheManager` extending `org.springframework.cache.transaction.AbstractTransactionSupportingCacheManager` leaving our `JpaTransactionManager` for our `entityManagerFactory` as it was ?

**Comment by carlspring** [comment]
@fuss86 : I believe that's what @sbespalov meant and I did incorrectly.

**Comment by sbespalov** [comment]
> @sbespalov can't we re-implement com.hazelcast.spring.cache.HazelcastCacheManager extending org.springframework.cache.transaction.AbstractTransactionSupportingCacheManager leaving our JpaTransactionManager for our entityManagerFactory as it was ?

yeah, we can try this way. But this is strange why guys from Hazelcast don't use `AbstractTransactionSupportingCacheManager` or even `AbstractCacheManager`.

**Comment by fuss86** [comment]
Great, I'll give it a try tomorrow.

**Comment by carlspring** [comment]
Work continued by @fuss86 in #691.

**Pull request closed without merging**